File: app/src/main/java/gateway/hitrontech/com/encryption/utils/FileUtils.java
package gateway.hitrontech.com.encryption.utils;

import android.os.Environment;
import java.io.File;

public class FileUtils {

  static final String ORIGIN = "origin";

  static final String TARGET = "target";

  static final String RESULT = "result";

  public static String getCachePath() {

    if (Environment.MEDIA_MOUNTED.equals(Environment.getExternalStorageState())
        || !Environment.isExternalStorageRemovable()) {
      return WhichContext.getInstance().getExternalCacheDir().getPath();
    } else {
      return WhichContext.getInstance().getCacheDir().getPath();
    }

  }

  public static String getResultPath() {
    File result = new File(getCachePath().replace("cache", "") + "/" + RESULT);
    if (!result.exists()) {
      result.mkdir();
    }
    return result.getPath();
  }


  public static String getOrigin() {
    return getResultPath() + "/" + ORIGIN;
  }

  public static String getTarget() {
    return getResultPath() + "/" + TARGET;
  }


  public static String getTargetXls() {
    return getResultPath() + "/" + TARGET + ".xls";
  }

}

File: app/src/main/java/gateway/hitrontech/com/encryption/utils/FileUtils.kt
package gateway.hitrontech.com.encryption.utils

import android.os.Environment
import java.io.File

object FileUtils {

    private val ORIGIN = "origin"

    private val TARGET = "target"

    private val RESULT = "result"

    val cachePath: String
        get() = if (Environment.MEDIA_MOUNTED == Environment.getExternalStorageState() || !Environment.isExternalStorageRemovable()) {
            WhichContext.instance!!.getExternalCacheDir()!!.getPath()
        } else {
            WhichContext.instance!!.getCacheDir().getPath()
        }

    val resultPath: String
        get() {
            val result = File(cachePath.replace("cache", "") + "/" + RESULT)
            if (!result.exists()) {
                result.mkdir()
            }
            return result.path
        }


    val origin: String
        get() = "$resultPath/$ORIGIN"

    val target: String
        get() = "$resultPath/$TARGET"


    val targetXls: String
        get() = "$resultPath/$TARGET.xls"

}
